content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,302 @@
+
+To learn more read the [AppArmor tutorial](/docs/tutorials/security/apparmor/). This work was done as a part of [KEP #24](https://github.com/kubernetes/enhancements/issues/24) lead by  [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+
+### Improved ingress connectivity reliability for Kube-proxy
+
+Kube-proxy improved ingress connectivity reliability is stable in v1.31. One of the common problems with load balancers in Kubernetes is the synchronization between the different components involved to avoid traffic drop. This feature implements a mechanism in kube-proxy for load balancers to do connection draining for terminating Nodes exposed by services of `type: LoadBalancer` and `externalTrafficPolicy: Cluster` and establish some best practices for cloud providers and Kubernetes load balancers implementations.
+
+To use this feature, kube-proxy needs to run as default service proxy on the cluster and the load balancer needs to support connection draining. There are no specific changes required for using this feature, it has been enabled by default in Kube-proxy since 1.30 and been promoted to stable in 1.31.
+
+For more details about this feature please visit the [Virtual IPs and Service Proxies documentation page](/docs/reference/networking/virtual-ips/#external-traffic-policy).
+
+This work was done as part of [KEP #3836](https://github.com/kubernetes/enhancements/issues/3836) by [SIG Network](https://github.com/kubernetes/community/tree/master/sig-network).
+    
+
+### Persistent Volume last phase transition time
+
+Persistent Volume last phase transition time feature moved to GA in v1.31. This feature adds a `PersistentVolumeStatus` field which holds a timestamp of when a PersistentVolume last transitioned to a different phase.
+With this feature enabled, every PersistentVolume object will have a new field `.status.lastTransitionTime`, that holds a timestamp of
+when the volume last transitioned its phase. This change is not immediate; the new field will be populated whenever a PersistentVolume is updated and first transitions between phases (`Pending`, `Bound`, or `Released`) after upgrading to Kubernetes v1.31.
+This allows you to measure time between when a PersistentVolume moves from `Pending` to `Bound`. This can be also useful for providing metrics and SLOs.
+
+For more details about this feature please visit the [PersistentVolume documentation page](/docs/concepts/storage/persistent-volumes/).
+
+This work was done as a part of [KEP #3762](https://github.com/kubernetes/enhancements/issues/3762) lead by  [SIG Storage](https://github.com/kubernetes/community/tree/master/sig-storage).
+
+## Highlights of features graduating to Beta
+
+_This is a selection of some of the improvements that are now beta following the v1.31 release._
+    
+### nftables backend for kube-proxy
```

Comment:
@danwinship can you review this section of the release blog?

Reply:
resolving this for a new comment, https://github.com/kubernetes/website/pull/47281/files#r1711552290